diagnosis/getkuberneteslogs.sh

```diff
@@ -23,6 +23,151 @@ function restore_ssh_config
+    else
+        azure_subscription_id=$(az account list --output json | jq -r '.[] | select(.isDefault==true) | .id')
+    fi
+    echo "Using subscription_id: $azure_subscription_id"
+}
+
+askName() {
+    echo ""
+    echo "Choose a name for your resource group, storage account and client"
+    echo "client. This is arbitrary, but it must not already be in use by"
+    echo "any of those resources. ALPHANUMERIC ONLY. Ex: mypackerbuild"
+    echo -n "> "
+    read meta_name
+}
+
+createResourceGroup() {
+    local location=$1
+    echo "==> Creating resource group"
+    az group create -n $meta_name -l $location
+    if [ $? -eq 0 ]; then
+        azure_group_name=$meta_name
+    else
+        echo "Error creating resource group: $meta_name"
+        return 1
+    fi
+}
+
+createStorageAccount() {
+
+    local location=$1 
+
+    echo "==> Creating storage account"
+    az storage account create --name $meta_name --resource-group $meta_name --location $location --kind Storage --sku Standard_LRS
+    if [ $? -eq 0 ]; then
+        azure_storage_name=$meta_name
+    else
+        echo "Error creating storage account: $meta_name"
+        return 1
+    fi
+}
+
+
+retryable() {
+    n=0
+    until [ $n -ge $1 ]
+    do
+        $2 && return 0
+        echo "$2 failed. Retrying..."
+        n=$[$n+1]
+        doSleep
+    done
+    echo "$2 failed after $1 tries. Exiting."
+    exit 1
+}
+
+requirements() {
+    found=0
+
+    azureversion=$(az --version)
+    if [ $? -eq 0 ]; then
+        found=$((found + 1))
+        echo "Found azure-cli version: $azureversion"
+    else
+        echo "azure-cli is missing. Please install azure-cli from"
+        echo "https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest"
+        echo "Alternatively, you can use the Cloud Shell https://docs.microsoft.com/en-us/azure/cloud-shell/overview right from the Azure Portal or even VS Code."
+    fi
+
+    jqversion=$(jq --version)
+    if [ $? -eq 0 ]; then
+        found=$((found + 1))
+        echo "Found jq version: $jqversion"
+    else
+        echo "jq is missing. Please install jq from"
+        echo "https://stedolan.github.io/jq/"
+    fi
+
+    if [ $found -lt 2 ]; then
+        exit 1
+    fi
+}
+
+
+function upload_logs()
+{   
+    local file_to_upload=$1    
+    local storage_acct=$2
+    local storage_key=$3
+
+    storage_key=$(az storage account keys list -g $metaname -n $metaname | jq '.[0].value ')
+    if [ "$storage_key" == "" ]; then
+        echo "Unable to retrieve the storage key for the stroage account $meta_name"
+        exit
+    fi 
+
+    CURRENTDATE=$(date +"%Y-%m-%d-%H-%M-%S-%3N")
+    container_name="AzureStack_KubernetesLogs_$CURRENTDATE"
+    blob_name="KubernetesLogs_$CURRENTDATE"
+    
+    az storage container create --name $container_name --account-name $meta_name --account-key $storage_key
+    if [ $? -ne 0 ]: then
+        echo "Error creating the container $container_name"
```

Comment:
I think `the` is not correct in this context. Maybe `Error creating blob container ...`